src/Build/BackEnd/Components/SdkResolution/SdkResolverLoader.cs

```diff
@@ -20,10 +22,27 @@ internal class SdkResolverLoader
         {
             // Always add the default resolver
             var resolvers = new List<SdkResolver> {new DefaultSdkResolver()};
+
+            // Add the NuGet-based SDK resolver unless its is disabled
+            if (!Traits.Instance.EscapeHatches.DisableNuGetSdkResolver)
+            {
+                try
+                {
+                    resolvers.Add(new NuGetSdkResolver());
+                }
+                catch (Exception e)
+                {
+                    loggingContext.LogWarning(string.Empty, new BuildEventFileInfo(location), "CouldNotLoadSdkResolver", e.Message);
+                }
+            }
+
             var potentialResolvers = FindPotentialSdkResolvers(
                 Path.Combine(BuildEnvironmentHelper.Instance.MSBuildToolsDirectory32, "SdkResolvers"));
 
-            if (potentialResolvers.Count == 0) return resolvers;
+            if (potentialResolvers.Count == 0)
+            {
+                return resolvers.Count == 1 ? resolvers : resolvers.OrderBy(t => t.Priority).ToList();
```

Comment:
Was this optimization noticeably faster?

Reply:
I left it in to keep existing functionality.  @AndyGerlicher had the optimization in there first, possibly by accident